source/frontend/src/features/mapSideBar/acquisition/tabs/expropriation/form8/Form8PaymentItemsSubForm.tsx

```diff
@@ -0,0 +1,205 @@
+import { FieldArray, FormikProps, useFormikContext } from 'formik';
+import { useState } from 'react';
+import { FaTrash } from 'react-icons/fa';
+import styled from 'styled-components';
+
+import { LinkButton, StyledRemoveLinkButton } from '@/components/common/buttons';
+import { FastCurrencyInput, Select } from '@/components/common/form';
+import { YesNoSelect } from '@/components/common/form/YesNoSelect';
+import GenericModal from '@/components/common/GenericModal';
+import { SectionField } from '@/components/common/Section/SectionField';
+import * as API from '@/constants/API';
+import useLookupCodeHelpers from '@/hooks/useLookupCodeHelpers';
+import { getCurrencyCleanValue, stringToBoolean } from '@/utils/formUtils';
+
+import { Form8FormModel, Form8PaymentItemModel } from './models/Form8FormModel';
+
+export interface IForm8PaymentItemsSubFormProps {
+  form8Id: number | null;
+  formikProps: FormikProps<Form8FormModel>;
+  gstConstantPercentage: number;
+}
+
+export const Form8PaymentItemsSubForm: React.FunctionComponent<IForm8PaymentItemsSubFormProps> = ({
+  form8Id,
+  formikProps,
+  gstConstantPercentage,
+}) => {
+  const { values, setFieldValue } = useFormikContext<Form8FormModel>();
+  const [showModal, setShowModal] = useState(false);
+  const [rowToDelete, setRowToDelete] = useState<number | undefined>(undefined);
```

Comment:
see below, should be able to use a closure to get rid of this. We should not be using state outside of Container* methods.

Reply:
updated